crates/precompile/src/bn128/arkworks.rs

```diff
@@ -0,0 +1,213 @@
+use super::{FQ2_LEN, FQ_LEN, G1_LEN, SCALAR_LEN};
+use crate::PrecompileError;
+use std::vec::Vec; // TODO: Remove this import, `cargo clippy --no-default-features` needs it to disambiguate
+
+use ark_bn254::{Bn254, Fq, Fq2, Fr, G1Affine, G1Projective, G2Affine};
+use ark_ec::{pairing::Pairing, AffineRepr, CurveGroup};
+use ark_ff::{One, PrimeField, Zero};
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+
+/// Reads a single `Fq` field element from the input slice.
+///
+/// Takes a byte slice and attempts to interpret the first 32 bytes as an
+/// elliptic curve field element. Returns an error if the bytes do not form
+/// a valid field element.
+///
+/// # Panics
+///
+/// Panics if the input is not at least 32 bytes long.
+#[inline]
+fn read_fq(input_be: &[u8]) -> Result<Fq, PrecompileError> {
+    // TODO: maybe assert input_be is 32 bytes, then change panic note to _exactly_ 32 bytes?
+
+    let mut input_le = [0u8; FQ_LEN];
+    input_le.copy_from_slice(input_be);
+
+    // Reverse in-place to convert from big-endian to little-endian.
+    input_le.reverse();
```

Comment:
arkworks uses little endian while eip1962 uses big endian, so we need to reverse the input